src/components/Header/index.tsx

```diff
@@ -1,77 +1,90 @@
+import { useSelector, useDispatch } from 'react-redux';
+import { coordinatesSelector, weatherAllSelector } from 'store/selectors';
+
 import { nanoid } from 'nanoid';
 // function
-import { weathers } from '../../store/from-service/action';
-//components
-import { Button } from '../../shared';
+import { getWeathers } from 'store/weather-service/action';
 // types
-import { IWeatherAll, IWeatherReducer } from '../../store/types';
+import { Daily } from 'core/types';
+// consts
+import { baseImgUrl, days } from 'core/consts';
+// components
+import { Button } from 'shared';
 // styles
 import s from './Header.module.scss';
 
-const HeaderTest = () => {
+const Header: FC = () => {
 	const dispatch = useDispatch();
-	const { weather }: IWeatherReducer = useSelector((state: IWeatherReducer) => state.weathers);
+	const all = useSelector(weatherAllSelector);
+	const { lat, lng } = useSelector(coordinatesSelector);
 
-	const baseImgUrl = 'http://openweathermap.org/img/wn/';
-	const [temp, setTemp] = useState(false);
+	const [temperatureValue, setTemperatureValue] = useState(false);
 	const [changeWeather, setChangeWeather] = useState(true);
 
 	useEffect(() => {
-		dispatch(weathers());
+		dispatch(getWeathers({ lat, lng }));
 	}, []);
 
-	const days = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
 	const countTemp = (data: number): string => {
-		if (!temp) {
-			return Math.round(data) + ' °C';
-		} else {
-			return (Math.round(data) * 9) / 5 + 32 + ' °F';
+		if (!temperatureValue) {
+			return `${Math.round(data)} °C`;
 		}
+
+		return `${(Math.round(data) * 9) / 5 + 32} °F`;
 	};
 
-	const { hourly } = weather as IWeatherAll;
-	const { daily } = weather as IWeatherAll;
 	return (
 		<div className={s.header}>
 			<div className={s.header__btns}>
 				<div className={s.header__view}>
 					<Button
 						variant="text"
 						title="Today"
-						days={true}
+						days
 						onClick={() => setChangeWeather(false)}
 						daysChange={!changeWeather}
 					/>
 					<Button
 						variant="text"
 						title="Week"
-						days={true}
+						days
 						onClick={() => setChangeWeather(true)}
 						daysChange={changeWeather}
 					/>
 				</div>
 				<div className={s.header__temperature}>
-					<Button variant="text" title="°C" tempChange={!temp} temp={true} onClick={() => setTemp(false)} />
-					<Button variant="text" title="°F" temp={true} tempChange={temp} onClick={() => setTemp(true)} />
+					<Button
+						variant="text"
+						title="°C"
+						tempChange={!temperatureValue}
+						temp
+						onClick={() => setTemperatureValue(false)}
+					/>
+					<Button
+						variant="text"
+						title="°F"
+						temp
+						tempChange={temperatureValue}
+						onClick={() => setTemperatureValue(true)}
+					/>
 				</div>
 			</div>
 			<div className={s.header__content}>
-				{(changeWeather ? daily || [] : hourly || []).slice(0, 8).map((item: any) => (
+				{(changeWeather ? all.daily || [] : all.hourly || []).slice(0, 8).map(temp => (
```

Comment:
all.daily is Array ? all.daily.slice : all?.daily?.slice

Reply:
done